packages/components/src/components/tooltip/tooltip.js

```diff
@@ -151,71 +177,183 @@ class Tooltip extends mixin(
+        //      moved to the tooltip. I'm assuming this is due to the debounce. Now that it's triggered on "click"
+        //      for keyboard users do we still want the debounce?
+        this.isHidden = state !== 'shown';
+        this.tooltip.element.setAttribute(
+          'aria-hidden',
+          this.isHidden.toString()
+        );
+        this.element.setAttribute('aria-expanded', (!this.isHidden).toString());
+
+        if (this._handleFocusinListener) {
+          this._handleFocusinListener = this.unmanage(
+            this._handleFocusinListener
+          ).release();
+        }
+
+        if (this._handleClickListener) {
+          this._handleClickListener = this.unmanage(
+            this._handleClickListener
+          ).release();
+        }
+
+        if (this._handleKeydownListener) {
+          this._handleKeydownListener = this.unmanage(
+            this._handleKeydownListener
+          ).release();
+        }
+
+        if (state === 'shown') {
+          this._hookCloseActions(this.element);
+          const focusableNode = this.tooltip.element.querySelector(
+            settings.selectorTabbable
+          );
+          if (focusableNode) {
+            focusableNode.focus();
+          } else {
+            this.tooltip.element.setAttribute('tabindex', '0');
+            this.tooltip.element.focus();
+          }
+
+          const hasFocusin =
+            'onfocusin' in this.element.ownerDocument.defaultView;
+          const focusinEventName = hasFocusin ? 'focusin' : 'focus';
+          this._handleFocusinListener = this.manage(
+            on(
+              this.element.ownerDocument,
+              focusinEventName,
+              this._handleFocusin,
+              !hasFocusin
+            )
+          );
+        } else {
+          this.element.focus();
+        }
+
+        // @question Do I need to pass any parameters to callback?
+        if (typeof callback === 'function') callback();
+      }
     );
   }
 
   /**
-   * Attaches event handlers to show/hide the tooltip.
+   * Attaches event handlers to show the tooltip.
    * @param {Element} element The element to attach the events to.
    * @private
    */
   _hookOn(element) {
-    const hasFocusin = 'onfocusin' in window;
-    const focusinEventName = hasFocusin ? 'focusin' : 'focus';
-    [focusinEventName, 'blur', 'touchleave', 'touchcancel'].forEach(name => {
+    this.manage(
+      on(
+        element,
+        'click',
+        event => {
+          const { relatedTarget, type } = event;
+          const hadContextMenu = this._hasContextMenu;
+          this._hasContextMenu = type === 'contextmenu';
+          this._debouncedHandleClick({
+            relatedTarget,
+            type,
+            hadContextMenu,
+            details: getLaunchingDetails(event),
+          });
+        },
+        false
+      )
+    );
+
+    if (this.element.tagName.toLowerCase() !== 'button') {
```

Comment:
`.tagName` is always upper case for HTML elements, so we can just compare `.tagName` with `BUTTON`?